src/overlay/overlay.scss

```diff
@@ -0,0 +1,376 @@
+// replace with primitive
+$primer-borderRadius-large: 0.75rem;
+
+.Overlay-hidden {
+  display: none !important;
+}
+
+.Overlay-visibilityHidden {
+  height: 0;
+  overflow: hidden;
+  visibility: hidden;
+  opacity: 0;
+}
+
+.Overlay {
+  display: flex;
+  min-width: 192px;
+  flex-direction: column;
+  background-color: var(--color-canvas-default);
+  border-radius: $primer-borderRadius-large;
+  box-shadow: var(--color-overlay-shadow);
+  opacity: 1;
+
+  &.Overlay--height-auto {
+    height: auto;
+  }
+
+  &.Overlay--height-xsmall {
+    height: 192px;
+  }
+
+  &.Overlay--height-small {
+    height: 256px;
+  }
+
+  &.Overlay--height-medium {
+    height: 320px;
+  }
+
+  &.Overlay--height-large {
+    height: 432px;
+  }
+
+  &.Overlay--height-xlarge {
+    height: 600px;
+  }
+
+  &.Overlay--width-auto {
+    width: auto;
+  }
+
+  &.Overlay--width-small {
+    min-width: 256px;
+  }
+
+  &.Overlay--width-medium {
+    min-width: 320px;
+  }
+
+  &.Overlay--width-large {
+    min-width: 480px;
+  }
+
+  &.Overlay--width-xlarge {
+    min-width: 640px;
+  }
+
+  &.Overlay--width-xxlarge {
+    min-width: 960px;
+  }
+
+  &.Overlay--motion-scaleFade {
+    @media screen and (prefers-reduced-motion: no-preference) {
+      animation: 200ms cubic-bezier(0.33, 1, 0.68, 1) 0s 1 normal none running motion--scaleFade;
+    }
+  }
+
+  @keyframes motion--scaleFade {
+    0% {
+      opacity: 0;
+      transform: scale(0.5);
+    }
+
+    100% {
+      opacity: 1;
+      transform: scale(1);
+    }
+  }
+}
+
+.Overlay-header {
+  z-index: 1;
+  display: flex;
+  flex-direction: column;
+
+  &.Overlay-header--divided {
+    padding-bottom: $spacer-2;
+    // stylelint-disable-next-line primer/box-shadow
+    box-shadow: inset 0 #{-$border-width} var(--color-border-default);
+  }
+
+  &.Overlay-header--large {
+    .Overlay-headerContentWrap {
+      .Overlay-titleWrap {
+        gap: $spacer-2;
+
+        .Overlay-title {
+          font-size: $h3-size;
+        }
+
+        .Overlay-description {
+          font-size: $body-font-size;
+        }
+      }
+    }
+  }
+
+  .Overlay-headerContentWrap {
+    display: flex;
+    align-items: flex-start;
+    gap: $spacer-2;
+    padding: $spacer-2 $spacer-2 0 $spacer-2;
+
+    .Overlay-actionWrap {
+      display: flex;
+      flex-direction: row;
+      gap: $spacer-2;
+    }
+
+    .Overlay-titleWrap {
+      display: flex;
+      padding: ($spacer-2 * 0.75) 0 ($spacer-2 * 0.75) $spacer-2;
+      flex-direction: column;
+      flex-grow: 1;
+      gap: $spacer-1;
+
+      .Overlay-title {
+        margin: 0;
+        font-size: $body-font-size;
+        font-weight: $font-weight-bold;
+      }
+
+      .Overlay-description {
+        margin: 0;
+        font-size: $font-size-small;
+        font-weight: $font-weight-normal;
+        color: var(--color-fg-muted);
+      }
+    }
+  }
+}
+
+// generic body content slot
+.Overlay-body {
+  flex-grow: 1;
+  padding: $spacer-3;
```

Comment:
Can we make the `Overlay-body`'s `padding-top` only appear in case there's a header divider as well? That'd reduce the vertical spacing between the overlay header and body a little:

| Currently | Proposal |
|--------|--------|
| ![Screen Shot 2022-04-26 at 15 06 30](https://user-images.githubusercontent.com/293280/165400461-04a1b111-ac64-4844-ae3c-3403f20914cf.png) | ![Screen Shot 2022-04-26 at 15 06 18](https://user-images.githubusercontent.com/293280/165400413-458b8b5a-64ed-4552-84e0-6a611c3e6c6e.png) | 

In case there's a divider, the current vertical padding remains relevant:

> ![Screen Shot 2022-04-26 at 15 07 56](https://user-images.githubusercontent.com/293280/165400690-1b7e7644-2e94-4cb6-bb9d-d63459bbe6fd.png)

For a scrolling body, here's how it could look, with and without a divider:

| With divider | Without divider |
|--------|--------|
| ![Screen Shot 2022-04-26 at 15 10 25](https://user-images.githubusercontent.com/293280/165401019-2e3c908a-2bad-4e67-ae46-005ab80acba0.png) | ![Screen Shot 2022-04-26 at 15 10 46](https://user-images.githubusercontent.com/293280/165401036-d6238726-abfe-47c5-82f2-f10fd64cc545.png) |

Reply:
I think I solved for this 🤞